### PR TITLE
BZ2103198_4_12:Add blocked upgrade restriction to vSphere CSI

### PR DESCRIPTION
**Pull request by lpettyjo**
We are not blocking upgrades to 4.11 when a 3rd-party vSphere CSI driver is detected due to feature parity issues with our driver vs 3rd-party drivers, but are adding the blocking for 4.13

PR to remove blocking is here: https://github.com/openshift/openshift-docs/pull/47526

Version(s): 4.12+

Issue: https://bugzilla.redhat.com/show_bug.cgi?id=2103198

Link to docs preview: http://file.rdu.redhat.com/lpettyjo/BZ2103198_4_12/storage/container_storage_interface/persistent-storage-csi-vsphere.html#vsphere-csi-driver-reqs_persistent-storage-csi-vsphere

PTAL: @bscott-rh, @gnufied, @gcharot

### CODE REVIEW

File: modules/persistent-storage-csi-vsphere-install-issues.adoc [hunk 1]
```diff
@@ -6,9 +6,9 @@
...
 
-{product-title} 4.10 clusters are still fully supported, and updates to z-stream releases of 4.10, such as 4.10.z, are not blocked, but you must correct this state by removing the third-party vSphere CSI Driver before updates to next major version of {product-title} can occur. Removing the third-party vSphere CSI driver does not require deletion of associated persistent volume (PV) objects, and no data loss should occur.
+{product-title} 4.12 clusters are still fully supported, and updates to z-stream releases of 4.12, such as 4.12.z, are not blocked, but you must correct this state by removing the third-party vSphere CSI Driver before updates to next major version of {product-title} can occur. Removing the third-party vSphere CSI driver does not require deletion of associated persistent volume (PV) objects, and no data loss should occur.
```
Comment (duanwei33) [inline modules/persistent-storage-csi-vsphere-install-issues.adoc:11]: TBH, I'm not sure the expected behavior in coming 4.12, but I think the update is ok to keep the consistency with previous releases.
CC @gcharot to take a look.

Comment (gcharot) [reply modules/persistent-storage-csi-vsphere-install-issues.adoc:11]: yes that's the envisioned behavior. we are planning to block 4.12 to 4.13 upgrades if we detect a third party CSI driver.

**Comment by bergerhoffer** [comment]
The enterprise-4.12 label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.11. And any PR going into `main` must also target the latest version branch (enterprise-4.12).

If the update in your PR does NOT apply to version 4.12 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

File: modules/persistent-storage-csi-vsphere-install-issues.adoc [hunk 2]
```diff
@@ -6,9 +6,9 @@
 [id="persistent-storage-csi-vsphere-install-issues_{context}"]
 = Removing a third-party vSphere CSI Operator Driver
 
-{product-title} 4.10 includes a built-in version of the vSphere Container Storage Interface (CSI) Operator Driver that is supported by Red Hat. If you have installed a vSphere CSI driver provided by the community or another vendor, upgrades will be prevented in a future release of {product-title}.
+{product-title} 4.12, and later, includes a built-in version of the vSphere Container Storage Interface (CSI) Operator Driver that is supported by Red Hat. If you have installed a vSphere CSI driver provided by the community or another vendor, updates to the next major version of {product-title}, such as 4.13, or later, might be disabled for your cluster.
```
Comment (gnufied) [inline modules/persistent-storage-csi-vsphere-install-issues.adoc:9]: Technically we do ship the builtin operator from 4.10 itself. So we shouldn't change versioning there I think.

Comment (gcharot) [reply modules/persistent-storage-csi-vsphere-install-issues.adoc:9]: yes agree it may confuse customers.

Comment (lpettyjo) [reply modules/persistent-storage-csi-vsphere-install-issues.adoc]: Fixed it.

**Pull request merged**